File: MergeStringsAlternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
 
         string s="";
         
         if(word1.size()==word2.size()){
              for(int i=0;i<word2.size();i++){
                  s+=word1[i];
                  s+=word2[i];
              }
          
         }
         if(word1.size()>word2.size()){
              for(int i=0;i<word2.size();i++){
                  s+=word1[i];
                  s+=word2[i];
              }
              int j=word2.size();
            while(j!=word1.size()){
              s+=word1[j];
              j++;
         }
         }
          if(word2.size()>word1.size()){
              for(int i=0;i<word1.size();i++){
                  s+=word1[i];
                  s+=word2[i];
              }
              int j=word1.size();
            while(j!=word2.size()){
              s+=word2[j];
              j++;
         }
         }

    return s;
         
       
       
        
    }
};

File: leetcode_solution/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        d=""
        l1=len(word1)
        l2=len(word2)
        mn=max(l1,l2)
        for i in range(mn):
            if(i<l1 and i<l2):
                d+=word1[i]
                d+=word2[i]
            elif(i<l1 and i>=l2):
                d+=word1[i]
            elif(i>=l1 and i<l2):
                d+=word2[i]
            else:
                continue
        return d            
            
        